### PR TITLE
Gen2 apriltag

### PR DESCRIPTION
**Pull request by csaba-luxonis**
WIP. Currently it runs on full frame with ~6 fps, with image manip downscaled it could be 30+ fps but it does not support RAW8 yet (crashes).

### CODE REVIEW

File: include/depthai/pipeline/datatype/AprilTagData.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    AprilTagData();
+    explicit AprilTagData(std::shared_ptr<RawAprilTags> ptr);
+    virtual ~AprilTagData() = default;
+
+    /**
+     * Retrieve configuration data for AprilTagData.
+     * @returns Vector of spatial location data, carrying spatial information (X,Y,Z)
+     */
+    std::vector<AprilTags>& getAprilTag() const;
```
Comment (themarpe) [inline include/depthai/pipeline/datatype/AprilTagData.hpp:30]: Lets leave this out, same as `ImgDetections` message.

File: src/pipeline/datatype/StreamPacketParser.cpp [hunk 2]
```diff
@@ -118,7 +122,15 @@ std::shared_ptr<RawBuffer> parsePacket(streamPacketDesc_t* packet) {
             break;
 
         case DatatypeEnum::SpatialLocationCalculatorConfig:
-            return parseDatatype<RawSpatialLocations>(jser, data);
+            return parseDatatype<RawSpatialLocationCalculatorConfig>(jser, data);
```
Comment (themarpe) [inline src/pipeline/datatype/StreamPacketParser.cpp:125]: Good catch!

**Review by themarpe — CHANGES_REQUESTED** [review]
And modifications that'll be required after changes are made to depthai-shared PR

**Comment by szabi-luxonis** [comment]
Update:

https://user-images.githubusercontent.com/60790666/151727465-55440c47-ad44-4e30-a195-8cf70963cdbf.mp4

Algorithm params are configurable to improve accuracy or performance.
https://github.com/luxonis/depthai-core/blob/gen2_apriltag/examples/AprilTag/apriltag.cpp#L39

**Pull request merged**